### PR TITLE
Fix/revpi 1144/clt src buster

### PR DESCRIPTION
**Pull request by zhan-kunbus**

### CODE REVIEW

File: collect_sources.sh [hunk 1]
```diff
@@ -58,6 +58,8 @@ EXCLUDE+='|pictory|revpi-(tools|webstatus)|revpi7'
 EXCLUDE+='|linux-4.9|raspberrypi-firmware|picontrol|revpi-firmware'
 # exclude RevolutionPi packages whose source code is fetched from GitHub
 EXCLUDE+='|python-snap7|snap7|revpimodio2'
+EXCLUDE+='|revpicommander|revpipyload|revpipycontrol|noderedrevpinodes-server'
```
Comment (iluminat23) [inline collect_sources.sh:62]: These packages will be repackaged with source packages included.

File: collect_sources.sh [hunk 2]
```diff
@@ -58,6 +58,8 @@ EXCLUDE+='|pictory|revpi-(tools|webstatus)|revpi7'
...
+EXCLUDE+='|nodejs'
```
Comment (iluminat23) [inline collect_sources.sh:63]: Why shoudl nodejs have no source package? It is a normal package from raspios.

**Review by iluminat23 — CHANGES_REQUESTED** [review]

**Review by iluminat23 — CHANGES_REQUESTED** [review]

File: collect_sources.sh [hunk 3]
```diff
@@ -82,7 +82,7 @@ wget -O wiringpi_$version.tar.gz \
 	https://github.com/WiringPi/WiringPi/archive/final_official_$version.tar.gz
 
 # fetch RevolutionPi sources
-wget -O linux.tar.gz https://github.com/RevolutionPi/linux/archive/revpi-4.9.tar.gz
+wget -O linux.tar.gz https://github.com/RevolutionPi/linux/archive/revpi-4.19.tar.gz
```
[the remark this answers is not in the document]
Comment (zhan-kunbus) [reply collect_sources.sh]: Is this one correct: https://github.com/RevolutionPi/linux/archive/refs/tags/raspberrypi-kernel_9.20210326-4.19.95+revpi1.tar.gz

File: collect_sources.sh [hunk 4]
```diff
@@ -82,7 +82,7 @@ wget -O wiringpi_$version.tar.gz \
 	https://github.com/WiringPi/WiringPi/archive/final_official_$version.tar.gz
 
 # fetch RevolutionPi sources
-wget -O linux.tar.gz https://github.com/RevolutionPi/linux/archive/revpi-4.9.tar.gz
+wget -O linux.tar.gz https://github.com/RevolutionPi/linux/archive/refs/tags/raspberrypi-kernel_9.20210326-4.19.95+revpi1.tar.gz
```
Comment (iluminat23) [inline collect_sources.sh:78]: This is useless as well. After a kernel Package upgrade this is wrong again!

**Review by iluminat23 — CHANGES_REQUESTED** [review]

File: collect_sources.sh (hunk 4 above)
Comment (zhan-kunbus) [reply collect_sources.sh:78]: should be fetch the correct version from apt-cache policy raspberrypi-kernel ?

Comment (iluminat23) [reply collect_sources.sh:78]: We need to fetch the kernel source matching the installed kernel package! The version can't be hardcoded in this script

**Review by iluminat23 — CHANGES_REQUESTED** [review]

File: collect_sources.sh (hunk 4 above)
Comment (zhan-kunbus) [reply collect_sources.sh:78]: So match the tag  (e.g. raspberrypi-kernel_9.20210326-4.19.95+revpi1) with the output of "apt show raspberrypi-kernel | grep Version: " should be a solution, right?

WARNING: apt does not have a stable CLI interface. Use with caution in scripts.

Version: 9.20210316-4.19.95+revpi1

Comment (zhan-kunbus) [reply collect_sources.sh:78]: or maybe we can provide the source package of raspberrypi-kernel

File: collect_sources.sh [hunk 5]
```diff
@@ -82,8 +82,11 @@ wget -O wiringpi_$version.tar.gz \
 	https://github.com/WiringPi/WiringPi/archive/final_official_$version.tar.gz
 
 # fetch RevolutionPi sources
-wget -O linux.tar.gz https://github.com/RevolutionPi/linux/archive/revpi-4.9.tar.gz
-wget -O piControl.tar.gz https://github.com/RevolutionPi/piControl/archive/master.tar.gz
+knl_version=$(dpkg-query --admindir $APTROOT/var/lib/dpkg -W \
+	-f='${source:Version}' raspberrypi-kernel || true)
+knl_tag="raspberrypi-kernel_$knl_version"
+wget -O linux.tar.gz "https://github.com/RevolutionPi/linux/archive/refs/tags/$knl_tag.tar.gz"
```
Comment (iluminat23) [inline collect_sources.sh:81]: Save the source with the correct version in the file name.

File: collect_sources.sh [hunk 6]
```diff
@@ -82,8 +82,11 @@ wget -O wiringpi_$version.tar.gz \
...
+wget -O piControl.tar.gz "https://github.com/RevolutionPi/piControl/archive/refs/tags/$knl_tag.tar.gz"
```
Comment (iluminat23) [inline collect_sources.sh:82]: Here as well.

**Review by iluminat23 — CHANGES_REQUESTED** [review]
We don't ship snap7 anymore and revpimodio has a source package now.

File: collect_sources.sh [hunk 7]
```diff
@@ -82,8 +82,11 @@ wget -O wiringpi_$version.tar.gz \
 	https://github.com/WiringPi/WiringPi/archive/final_official_$version.tar.gz
 
 # fetch RevolutionPi sources
-wget -O linux.tar.gz https://github.com/RevolutionPi/linux/archive/revpi-4.9.tar.gz
-wget -O piControl.tar.gz https://github.com/RevolutionPi/piControl/archive/master.tar.gz
+knl_version=$(dpkg-query --admindir $APTROOT/var/lib/dpkg -W \
+	-f='${source:Version}' raspberrypi-kernel || true)
+knl_tag="raspberrypi-kernel_$knl_version"
+wget -O "linux_$knl_tag.tar.gz" "https://github.com/RevolutionPi/linux/archive/refs/tags/$knl_tag.tar.gz"
+wget -O "piControl_$knl_tag.tar.gz" "https://github.com/RevolutionPi/piControl/archive/refs/tags/$knl_tag.tar.gz"
```
Comment (iluminat23) [inline collect_sources.sh:82]: "linux_$knl_tag.tar.gz" => "linux-$knl_tag.tar.gz"
"piControl_$knl_tag.tar.gz" => "piControl-$knl_tag.tar.gz"

File: collect_sources.sh [hunk 8]
```diff
@@ -49,15 +49,11 @@ EXCLUDE='realvnc-vnc|oracle-java8-jdk'
 # exclude Raspbian packages with missing source code
 EXCLUDE+='|nodered|wiringpi'
 # exclude binary-only RevolutionPi packages
-EXCLUDE+='|logi-rts|logiclab|piserial|procon-web-iot|teamviewer-revpi'
-# exclude binary-only RevolutionPi packages
-EXCLUDE+='|pimodbus-master|pimodbus-slave'
+EXCLUDE+='|logiclab|piserial|procon-web-iot|teamviewer-revpi|revpi-modbus'
```
Comment (iluminat23) [inline collect_sources.sh:57]: I think logiclab and procon-web-iot are also not part of buster.
@pavlic244?

**Review by iluminat23 — CHANGES_REQUESTED** [review]

File: collect_sources.sh [hunk 9]
```diff
@@ -28,6 +28,10 @@ mkdir -p $APTROOT/var/cache/apt/archives/partial
 	| (cd $APTROOT ; tar xf -)
 (cd / ; tar cf - /usr/lib/apt)	\
 	| (cd $APTROOT ; tar xf -)
+
+# no source package is provided by nodesource.list
+rm -f $APTROOT/etc/apt/sources.list.d/nodesource.list
```
Comment (iluminat23) [inline collect_sources.sh:33]: This is not what i meant. You are modifying the image. This is not ok. After the collec_sources script has been run the image must be the same as before.

**Review by iluminat23 — CHANGES_REQUESTED** [review]

File: collect_sources.sh (hunk 9 above)
Comment (zhan-kunbus) [reply collect_sources.sh:33]: IMAGEDIR point to the image, APTROOT is just /tmp/apt.xxx on host

Comment (iluminat23) [reply collect_sources.sh:33]: Yes, you are right!

File: collect_sources.sh [hunk 10]
```diff
@@ -47,17 +47,13 @@ fetch_deb_src() {
 # exclude binary-only Raspbian packages
 EXCLUDE='realvnc-vnc|oracle-java8-jdk'
```
Comment (iluminat23) [inline collect_sources.sh:53]: These two are not part of raspbian/raspios anymore. At least I can't find them.

File: collect_sources.sh [hunk 11]
```diff
@@ -47,17 +47,13 @@ fetch_deb_src() {
...
 # exclude Raspbian packages with missing source code
-EXCLUDE+='|nodered|wiringpi'
+EXCLUDE+='|nodered|wiringpi|nodejs'
```
Comment (iluminat23) [inline collect_sources.sh:55]: In the comment you wrote that nodejs was removed, not added

**Review by iluminat23 — CHANGES_REQUESTED** [review]

File: collect_sources.sh (hunk 8 above)
Comment (zhan-kunbus) [reply collect_sources.sh]: I removed both

File: collect_sources.sh (hunk 10 above)
Comment (zhan-kunbus) [reply collect_sources.sh]: realvnc-vnc is source of package: realvnc-vnc-server, it is kept
the oracle-java8-jdk removed

**Review by iluminat23 — APPROVED** [review]

**Pull request merged**